test/conformance/main_test.go

```diff
@@ -50,3 +57,53 @@ func TestMain(m *testing.M) {
 		return m.Run()
 	}())
 }
+
+func addSourcesInitializers(){
+	sourcesTestRunner.AddComponentSetupClientOption(
```

Comment:
I wonder, could the `testlib.ComponentsTestRunner` expose a method to add a retrieve a "stimulate" function too (you don't need to do it here, i'm just trying to understand if it's possible)? Does it makes sense

Reply:
lets discuss in the brainstorming